### DISCUSSION TITLE
Number of remaining quests in a given area?

### DISCUSSION THREAD

**Discussion by rmaree** [streetcomplete/StreetComplete#3499] [Ideas]
Hello,
Fantastic project !
Do you see a programmatic way to estimate the number of remaining quests (let say: all of them or a user selected subset of quest types) in a given area ? 
In addition to providing a way to evaluate progress and remaining effort in a given zone, i see this as a way to estimate the number of people+hours needed to organize successful mapathons using StreetComplete in given area/cities. I'm quite motivated to organize such events but it's not easy to estimate the scale of it.

**Comment by matkoniecz** [comment]
> needed to organize successful mapathons using StreetComplete in given area/cities

Note that mapathon that collected say 100 new addresses or deleted 20 no longer existing shops, or collected 100 opening hours is already successful, no matter how much data remained!

And getting to zero would take massive effort in well mapped areas.

---------------

https://streetcompleteness.haukauntrie.de/ is a proprietary software made by someone (note that solving one quest can cause other to become eligible)

In general one may use Overpass for that, for example http://overpass-turbo.eu/s/1cV9 queries for shops in Berlin without opening hours and http://overpass-turbo.eu/s/1cVc shows summary statistics.

You can also look into logs of StreetComplete as it is downloading data.

**Comment by smichel17** [comment]
Here's my best solution:

1. ONE TIME ONLY:
    - Enable Developer mode on your phone
    - Enable USB debugging on your phone
    - Install `adb` on your computer (by installing [Android Studio](https://developer.android.com/studio/) or just the  [platform tools](https://developer.android.com/studio/releases/platform-tools))
2. Clear all StreetComplete data
3. Turn off location services
4. Open StreetComplete, scroll to the area where you want to organize a mapathon and use manual download until the area you want to map (and nothing else!) is all downloaded.
5. Connect your phone to your computer via usb
6. Run this command in a terminal on your computer:
   ```sh
   adb shell run-as de.westnordost.streetcomplete 'sqlite3 /data/data/de.westnordost.streetcomplete/databases/streetcomplete_v2.db "SELECT COUNT(*) FROM osm_quests;"'
   ```
   If you are using the debug version instead (which I'd recommend, since it can be installed alongside the regular version, so you can avoid clearing data on your regular install), then the command is
   ```sh
   adb shell run-as de.westnordost.streetcomplete.debug 'sqlite3 /data/data/de.westnordost.streetcomplete.debug/databases/streetcomplete_v2.db "SELECT COUNT(*) FROM osm_quests;"'
   ```
    - Only tested on linux. Should work identically on mac. Windows might need some slight tweaking.

The output of the command will be the number of quests SC currently found. It won't be perfect, as @matkoniecz noted, because some quests won't appear until you answer a different quest, but it'll give you a good ballpark number.

We can definitely refine this process (e.g. improve the query so it won't require clearing all SC data first).
